ManageAxes.js

```diff
@@ -161,3 +170,79 @@ ManageAxes.resetAxes = async function()
+        return;
+    }
+
+    // get the bounding box of the editing history and its upper and lower bounds
+    const editingBBox = await WSM.APIGetBoxReadOnly(editingHistoryId);
+
+    // make a translation transform from the geometry to the world origin
+    const vec3d = await WSM.Geom.Vector3d(-(editingBBox.lower.x + editingBBox.upper.x) / 2, -(editingBBox.lower.y + editingBBox.upper.y) / 2, -editingBBox.lower.z);
+    const geomTransf3d = await WSM.Transf3d.MakeTranslationTransform(vec3d);
+    // get the inverse transform for use later
+    const inverseGeomTransf3d = await WSM.Transf3d.Invert(geomTransf3d);
+
+    // check if the transform from the geom to the world origin is null
+    const isGeomTransf3dNull = await WSM.Vector3d.IsNull(vec3d)
+    if (isGeomTransf3dNull) {
+        // show a message that nothing was changed
+        await FormIt.UI.ShowNotification("The origin is already centered below this geometry.\nNo changes were made.", FormIt.NotificationType.Information, 0);
+
+        // no need to go any further
+        return;
+    }
+
+    // if we got this far, changes will be made to the model, so start a new undo state
+    await FormIt.UndoManagement.BeginState();
+
+    // get all the non-owned objects in this history
+    const objectIds = await WSM.APIGetAllNonOwnedReadOnly(editingHistoryId);
+    // and move them to the world origin
+    await WSM.APITransformObjects(editingHistoryId, objectIds, geomTransf3d);
+
+    // get all the instances of this history
+    const instancesOfHistory = await WSM.APIGetAllAggregateTransf3dsReadOnly(editingHistoryId, 0);
+
+    // for each instance of this history...
+    for (let i = 0; i < instancesOfHistory.paths.length; i++) {
+        // how many levels "deep" is this instance from history 0
+        const historyDepthIndex = instancesOfHistory.paths[i].ids.length - 1;
+        const instanceObjectHistoryId = instancesOfHistory.paths[i].ids[historyDepthIndex];
+
+        // get the instance's transform and the inverse
+        const instanceTransf3d = instancesOfHistory.transforms[i];
+        const inverseInstanceTransf3d = await WSM.Transf3d.Invert(instanceTransf3d);
+
+        // multiply instance transform, instance inverse transform, and geom inverse transform
+        const newTransf3dPartial = await WSM.Transf3d.Multiply(inverseGeomTransf3d, inverseInstanceTransf3d);
+        const newTransf3dFinal = await WSM.Transf3d.Multiply(instanceTransf3d, newTransf3dPartial);
+
+        // transform the instance by the final transf3d
+        await WSM.APITransformObject(instanceObjectHistoryId.History, instanceObjectHistoryId.Object, newTransf3dFinal);
+    }
+
+    // reset the local origin (this gets moved along with the non-owned objects)
+    const defaultLCS = await WSM.Geom.MakeRigidTransform(await WSM.Geom.Point3d(0, 0, 0), await WSM.Geom.Vector3d(1, 0, 0), await WSM.Geom.Vector3d(0, 1, 0), await WSM.Geom.Vector3d(0, 0, 1));
```

Comment:
`const defaultLCS = await WSM.Transf3d.Transf3d()` will make the identity transform. No need to make it explicitly